### ISSUE TITLE
Can't get `validate` prop to work

### ISSUE THREAD

**Issue by triketora** [react-tools/react-form#251]
(Not sure if this is a bug or just me not knowing how to use the API.) I have two problems with `validate`. I can't get `validate` to work on 1. the `Form` itself, or 2. on a field.

My code looks something like this:
```javascript
class MyForm extends Component {

  handleValidate(values) {
    debugger; // never triggers
  }

  render() {
    return (
      <Form
        defaultValues={this.state.data}
        validateOnMount={true}
        validate={this.handleValidate}
        onSubmit={this.handleSubmit}
        getApi={this.setFormApi}
      >
      {formApi => 
         <form onSubmit={formApi.submitForm}>
           <Text field="name" id="name" validate={validateRequired} />
           ...
```

The `Form` level `validate` never gets called, and the `Text` field level `validate` causes this error:
```
Warning: Unknown prop `validate` on <input> tag. Remove this prop from the element. For details, see https://fb.me/react-unknown-prop
```

Any ideas? 

(I am on react-form version 3.0.1.)

**Comment by mjhm** [comment]
For a work around try 
```
...
       {formApi => 
         <form onSubmit={(e) => {e.preventDefault(); return formApi.submitForm(e)}}>
           <Text field="name" id="name" validate={validateRequired} />
           ...
```

I have a PR for a fix at https://github.com/react-tools/react-form/pull/250

The warning might be a separate issus.

**Comment by joepuzzo** [comment]
Hmm something must be wrong with your code.. i have code sandbox https://codesandbox.io/s/n4xr6w57wj that has this working I would try playing with the sandbox.

**Comment by joepuzzo** [comment]
@mjhm How is prevent default related to this?

**Comment by triketora** [comment]
Baffling. I've tried @mjhm's workaround but it doesn't work, and from what I can tell my code is setup the same way as what @joepuzzo put in the sandbox (I adapted my code to look even more similar to it), and no errors are shown. 

I've also verified that all my package versions are the same (for react, react-dom, etc.)

**Comment by mjhm** [comment]
@joepuzzo I was guessing that the form was submitting via the standard DOM `<form>` flow before running the validate function.  It seems I was wrong.

**Comment by joepuzzo** [comment]
cool and yeah it looks like all your pull request does is flip the logic of the prop.. not sure we need that

**Comment by triketora** [comment]
I fixed my problem by uninstalling react-form and re-installing it. I think the versions got messed up when I was trying to fix this problem: https://github.com/react-tools/react-form/issues/245. I don't remember exactly what I did but assuming I followed the instructions in that thread, I must have rolled back to 2.x, but npm kept thinking I was on 3.0.1. 

Thanks @mjhm and @joepuzzo for looking into this!
